fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -262,13 +262,50 @@ private static void setDeprecated(DescriptorProtos.FieldDescriptorProto.Builder
      * @return the resulting records descriptor
      */
     @Nonnull
-    public static Descriptors.FileDescriptor addDefaultUnionIfMissing(Descriptors.FileDescriptor fileDescriptor) {
+    public static Descriptors.FileDescriptor addDefaultUnionIfMissing(@Nonnull Descriptors.FileDescriptor fileDescriptor) {
         if (MetaDataProtoEditor.hasUnion(fileDescriptor)) {
             return fileDescriptor;
         }
         DescriptorProtos.FileDescriptorProto fileDescriptorProto = fileDescriptor.toProto();
         DescriptorProtos.FileDescriptorProto.Builder fileBuilder = fileDescriptorProto.toBuilder();
-        fileBuilder.addMessageType(MetaDataProtoEditor.createDefaultUnion(fileDescriptor));
+        fileBuilder.addMessageType(createDefaultUnion(fileDescriptor));
+        try {
+            return Descriptors.FileDescriptor.buildFrom(fileBuilder.build(), fileDescriptor.getDependencies().toArray(new Descriptors.FileDescriptor[0]));
+        } catch (Descriptors.DescriptorValidationException e) {
+            throw new MetaDataException("Failed to add a default union", e);
+        }
+    }
+
+    /**
+     * Creates a default union descriptor for the given file descriptor if missing.
+     *
+     * <p>
+     * It will first create a default union based on the union of the stored meta-data. It will then add the new record
+     * types to the union. This will preserve the order of the fields in the union.
```

Comment:
If the other Javadoc comments get reworded, I think this should be as well.

Reply:
which java doc?

Reply:
The Javadoc on `saveRecordMetaData` or `updateRecords`.

Reply:
done